### PR TITLE
chore: rebuild sinker to address jan-2024 glibc updates for debian bookworm

### PR DESCRIPTION
**Pull request by eumoh1601**
related to: https://github.com/influxdata/tubernetes/issues/1567

changes:

rebuild sinker to address jan-2024 glibc updates for debian bookworm

#Remove setuid/setgid bits from executables. Also If the path is `/proc`, don't descend into it. This effectively excludes `/proc` from the search.
`RUN find / -path /proc -prune -o -type f \( -perm -4000 -o -perm -2000 \) -exec chmod a-s {} \;`

### CODE REVIEW

**Review by marcdavoli — DISMISSED** [review]

File: Dockerfile [hunk 1]
```diff
@@ -18,12 +18,24 @@ RUN cargo build --release --bin sinker
...
+
+# Create a dedicated user and group for the application
+RUN groupadd --gid 1500 sinker \
     && useradd --uid 1500 --gid sinker --shell /bin/bash --create-home sinker
 
+# RUN apt update \
+#     && apt install --yes ca-certificates libssl3 --no-install-recommends \
+#     && rm -rf /var/lib/{apt,dpkg,cache,log} \
+#     && groupadd --gid 1500 sinker \
+#     && useradd --uid 1500 --gid sinker --shell /bin/bash --create-home sinker
```
Comment (marcdavoli) [inline Dockerfile:37]: Can you elaborate a little bit on why these changes are necessary?

**Comment by eumoh1601** [comment]
Updated the newly created image tested in k8s pods. Sinker pods seems to be running as usual.

**Review by marcdavoli — APPROVED** [review]
As discussed, let's get Jamie to take a look before we merge.

**Review by jdstrand — CHANGES_REQUESTED** [review]

File: Dockerfile (hunk 1 above)
Comment (jdstrand) [reply Dockerfile:37]: > That said, the addition of -path /proc -prune causes nothing to match.

CORRECTION: I did something wrong when I tested this. I can see now that `find / -path /proc -prune -o -type f \( -perm -4000 -o -perm -2000 \) -ls` does match stuff. This might be a refinement you'd still be interested in though:
```
# Remove setuid/setgid bits from executables as a hardening measure so non-root processes can't escalate.
RUN find / \( -path /dev -o -path /proc -o -path /sys \) -prune -o -type f \( -perm -4000 -o -perm -2000 \) -exec chmod a-s {} \;
```

**Review by jdstrand — APPROVED** [review]
I'm going to approve as is (now that I corrected myself), but please consider the couple things I mentioned.

**Comment by jdstrand** [comment]
@eumoh1601 thanks for working on this! :)
